internal/ipmi/ipmi.go

```diff
@@ -377,3 +384,70 @@ func (i *Ipmi) ReadUsers(ctx context.Context) (users []map[string]string, err er
 
 	return users, err
 }
+
+// SolInfo gets the status of the serial-over-lan interface
+func (i *Ipmi) SolInfo(ctx context.Context) (info string, err error) {
+	output, err := i.run(ctx, []string{"sol", "info"})
+	if err != nil {
+		return "", fmt.Errorf("%v: %v", err, output)
+	}
+
+	fmt.Println(output)
+	fmt.Println(err)
+
+	return output, nil
+}
+
+// SolActivate activates and connects to the serial-over-lan interface.
+// It needs stdin, or the connection will be immediately dropped.
+func (i *Ipmi) SolActivate(ctx context.Context, stdin ...byte) (output string, err error) {
+	output, err = i.run(ctx, []string{"sol", "activate"}, stdin...)
+	if err != nil {
+		return output, fmt.Errorf("%v: %v", err, output)
+	}
+
+	if strings.HasPrefix(output, "[SOL Session operational.") {
```

Comment:
Is this string consistent across vendors, models (atleast the ones we care about)?

Reply:
I tested this string and the cipher strings across g3 plans with hardware from dell, asrockrack, smc, and gigabyte. All of them matched. The tests using these features will be deployed more widely against the fleet, so if we find servers with different behavior, we can collect that info, but so far they all seem to match.